concepts/modularization/README.md

```diff
@@ -0,0 +1,170 @@
+
+[Operator bundle](https://olm.operatorframework.io/docs/tasks/creating-operator-bundle/#operator-bundle) is a container image that stores Kubernetes manifests and metadata associated with an operator. A bundle is meant to represent a specific version of an operator on cluster.
+Operator bundle contains ClusterServiceVersion resource that describes operator version and installation descriptor:
+https://olm.operatorframework.io/docs/tasks/creating-operator-manifests/#writing-your-operator-manifests
+
+OLM provides also abstraction for operator catalog with release channels. What can solve the problem with modules discovery for users.
+
+## Own solution
+OCM is a simple descriptor and requires additional structures to apply it for installing, configuring and updating Kyma modules (operators). OLM addresses release channels and operator catalogs but we need to adjust it to the model with operators running centrally and remote subscriptions. We need to decide if we build our own solution or reuse existing software. 
+We need the representation of the component in the control plane cluster. The idea is to create own custom resource that can have OCM component descriptor embedded. The ModuleTemplate custom resource would contain:
+- release channel
+- CRD for the module (the one managed by module operator)
+- deployment of the operator (Kubernetes manifests / helm chart)
+
+
+# Component submission process
+To submit a new component to Kyma you need to prepare a component descriptor (with all related resources). 
+
+Component validation should be automated by governance jobs that check different aspects like:
+- operator CRD validation (e.g. status format)
+- exposing metrics
+- proper logging format
+- ...
+
+Governance jobs should be owned by teams (no shared responsibility)
+
+Apart from technical quality, Kyma components should fulfill other standards (24/7 support, documenting micro-deliveries, etc)
+
+# Operator based component management
+
+`kyma-operator` is a meta operator responsible for installing and updating component operators. It is similar to [Operator Lifecycle Manger](https://olm.operatorframework.io/), but it can work in 2 modes: 
+- central mode (managing thousands of clusters) 
+- single cluster mode (local installation)
+
+## Regular (local) component operators
+Each Kyma component has to provide the operator that manages the component lifecycle (installing, updating, configuring, deleting). The component provider (team) has to prepare one custom resource (ComponentDescriptor) that specifies how to install the operator (deployment yaml) and how to configure it (operator CRD, default values). 
+The complexity of managing installation in many clusters or using configured release channels is handled by kyma-operator and component providers can ship regular kubernetes operators. Such regular operators are installed in the target cluster (where components should be installed). 
```

Comment:
We should align on a common spelling for the most important terms - like "Kyma Operator" and "Manifest Operator" etc. Capitalize it? Format it as code like in line 126? No need to do that now, but we'll need it eventually.

```suggestion
The complexity of managing installation in many clusters or using configured release channels is handled by `kyma-operator`. Component providers can ship regular Kubernetes operators. Such regular operators are installed in the target cluster where components should be installed. 
```